prepare.sh

```diff
@@ -93,10 +93,72 @@ install_command() {
 
 install_dependencies() {
     # Copy gitlab-runner binary from the server into the container
-    if [ -x /usr/local/bin/gitlab-runner ]; then
-        podman cp --pause=false /usr/local/bin/gitlab-runner "$CONTAINER_ID":/usr/bin/gitlab-runner
-    else
-        podman cp --pause=false /usr/bin/gitlab-runner "$CONTAINER_ID":/usr/bin/gitlab-runner
+    local RUNNER_BINARY
+    local RUNNER_BINARY_PREFIX
+    local RUNNER_BINARY_TMP
+
+    # First check some predefined paths
+    for RUNNER_BINARY_PREFIX in /usr{/local,}/bin
+    do
+      RUNNER_BINARY="${RUNNER_BINARY_PREFIX}/gitlab-runner"
+      if [ -x "${RUNNER_BINARY}" ]
+      then
+        break
+      fi
+    done
+
+    # If unsuccessful, check shell PATHs
+    if [ ! -x "${RUNNER_BINARY}" ]
+    then
+      RUNNER_BINARY="$(type -p gitlab-runner || true)"
+    fi
+
+    # As a last resort, download binary...
+    if [ ! -x "${RUNNER_BINARY}" ]
+    then
+      # ... to temporary directory
+      RUNNER_BINARY_TMP=$(mktemp --directory --tmpdir="${CACHE_DIR}")
```

Comment:
Does this need quoting around the subshell? You have it on line 113 but not here.

Reply:
One reason to use `shellcheck` as part of the CI

https://github.com/koalaman/shellcheck

Reply:
@JohnVillalovos Thanks for the tip. I am aware of shellcheck and use it already. shellcheck didn't report any errors, but I am not sure if that is a false negative or if quoting here really has no effect.

https://github.com/jonasbb/podman-gitlab-runner/blob/cddcf62cd6a1dc651f51e07c8bcc8106764f37bc/prepare.sh#L4

I look into adding a couple of automatic checks.

Reply:
> I look into adding a couple of automatic checks.

Almost always a good idea 👍

Reply:
They are [not needed](https://unix.stackexchange.com/a/97569) in bash, but since I've been trying to make a habit of quoting everything in shell scripts, it's likely I've either plain forgotten or was thrown by the quotes inside the subshell.
Anyway, the quotes should've been there, I've tested that they don't interfere with the inner quotes and I'll force-push them right away.

Reply:
Thanks for checking :)